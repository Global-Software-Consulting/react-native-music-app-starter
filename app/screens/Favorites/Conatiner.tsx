import React, { useState, useEffect } from 'react';
import { View, FlatList, TouchableHighlight } from 'react-native';
import { Text } from 'react-native-paper';
import { useDispatch, useSelector } from 'react-redux';
import useStyles from './styles';
import MusicCard from '../../components/Music/MusicCard';
import Header from '../../components/Header';
import { useNavigation } from '@react-navigation/native';
import { favoriteListRequest } from '../../store/actions/appActions';
import { IAppState } from '../../models/reducers/app';
import {ILoading} from '../../models/reducers/loading';

interface IState {
  appReducer: IAppState;
  loadingReducer: ILoading;
}
interface Itrack {

  id: string,
  url: string,
  title: string,
  artist: string,
  artwork: string,
  album: string,
  duration: number,
}
const Favorite: React.FC<any> = (props): JSX.Element => {
  const styles = useStyles();
  const navigation = useNavigation();
  const dispatch = useDispatch();
  const favoriteList = useSelector((state:IState) => state.appReducer.favoriteList);
  const removeFavorites = (id: any) => {
    let data = favoriteList?.filter((element: any) => element.id != id)
    dispatch(favoriteListRequest(data));
  };
  const favoriteRenderItem = ({ item }: any) => (
    <TouchableHighlight
      key={item}
      underlayColor='gray'
      onPress={() => { removeFavorites(item.id) }}
    >
      <View style={styles.Musiccontainer}>
        <MusicCard
          name={item.title}
          model={item.artist}
          img={item.artwork}
        />

      </View>
    </TouchableHighlight>
  );
  console.log("favlisttt:", favoriteList);

  return (

    <View style={styles.container}>

      <Header title="Liked Songs" />
      {favoriteList?.length > 0 ? (

        <FlatList
          contentContainerStyle={{ alignSelf: 'flex-start' }}
          showsVerticalScrollIndicator={false}
          showsHorizontalScrollIndicator={false}
          numColumns={2}
          data={favoriteList}
          keyExtractor={(item) => item.id}
          renderItem={favoriteRenderItem}

        />
      ) : (
        <View style={styles.container}>
          <Text style={styles.model}>No Favorities Available</Text>
        </View>

      )}

    </View>

  );
};

export default Favorite;
