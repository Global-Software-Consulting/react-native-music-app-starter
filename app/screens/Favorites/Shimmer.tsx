import React, { useState } from 'react';
import { View, FlatList, RefreshControl, ScrollView } from 'react-native';
import { tracks } from '../../data/tracks';
import useStyles from './styles';
import MusicCardShimmer from '../../components/Music/MusicCardShimmer';
import Header from '../../components/Header';
interface Itrack {
    id: string;
    url: string;
    title: string;
    artist: string;
    artwork: string;
    album: string;
    duration: number;
}
const FavoriteShimmer: React.FC<any> = () => {
    const Track: Itrack[] = tracks;
    const styles = useStyles();
    const [refreshing, setRefreshing] = useState<boolean>(false);
    const wait = (timeout: number) => {
        return new Promise((resolve) => setTimeout(resolve, timeout));
    };
    const onRefresh = React.useCallback(() => {
        setRefreshing(true);
        wait(2000).then(() => setRefreshing(false));
    }, []);
    const shimmerRenderItem = ({ item }: any) => (
        <View style={styles.Musiccontainer}>
            <MusicCardShimmer name={item.title} model={item.album} img={item.artwork} />
        </View>
    );
    return (
        <>
            <View style={styles.container}>
                <ScrollView
                    refreshControl={
                        <RefreshControl refreshing={refreshing} onRefresh={onRefresh} />
                    }>
                    <Header title="Liked Songs" />

                    <FlatList
                        contentContainerStyle={{ alignSelf: 'flex-start' }}
                        showsVerticalScrollIndicator={false}
                        showsHorizontalScrollIndicator={false}
                        numColumns={2}
                        data={Track}
                        keyExtractor={(item) => item.id}
                        renderItem={shimmerRenderItem}
                    />
                </ScrollView>
            </View>
        </>
    );
};

export default FavoriteShimmer;
