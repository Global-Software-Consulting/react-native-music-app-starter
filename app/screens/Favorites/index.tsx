import React, { useState, useEffect } from 'react';
import { View, ScrollView, StatusBar, TouchableOpacity } from 'react-native';
import LeftArrowIcon from 'react-native-vector-icons/MaterialIcons';
import AppHeader from '../../components/AppHeader';
import { useNavigation } from '@react-navigation/native';
import { useDispatch, useSelector } from 'react-redux';
import useStyles from './styles';
import { useTranslation } from 'react-i18next';
import { StackNavigationProp } from '@react-navigation/stack';
import { DrawerParamList } from '../../navigation/DrawerNavigator';
import FavoriteShimmer from './Shimmer';
import Favorite from './Conatiner';
import { AppState } from '../../models/reducers/app';
import { Loading } from '../../models/reducers/loading';

interface IState {
  appReducer: AppState;
  loadingReducer: Loading;
}
const Favorites: React.FC = () => {
  const dispatch = useDispatch();
  const isLoading = useSelector((state: IState) => state.loadingReducer.isLoginLoading);

  const [refreshing, setRefreshing] = useState<boolean>(false);
  const { t, i18n } = useTranslation();
  type homeScreenProp = StackNavigationProp<DrawerParamList, 'Home'>;
  const navigation = useNavigation<homeScreenProp>();
  const styles = useStyles();
 
  useEffect(() => {
    onRefresh();
  }, []);

  const onRefresh = () => {
    // getFavoriteList();
    if (isLoading) {
      <FavoriteShimmer />;
    } else {

    }
  };

  return (



    <View style={styles.container}>
      <AppHeader
        renderLeft={
          <TouchableOpacity
            style={{ paddingHorizontal: 20 }}
            onPress={() => {
              navigation.navigate('Home');
            }}>
            <LeftArrowIcon
              name="keyboard-arrow-left"
              style={styles.icon}
              size={30}

            />
          </ TouchableOpacity>}

      />
      {isLoading ? (
        <FavoriteShimmer />
      ) : (
        <Favorite />
      )}
    </View>


  );
};

export default Favorites;
