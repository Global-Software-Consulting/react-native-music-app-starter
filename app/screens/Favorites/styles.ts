import { StyleSheet } from 'react-native';

import { widthPercentageToDP as wp, heightPercentageToDP as hp } from 'react-native-responsive-screen';
import { useTheme } from 'react-native-paper';

export const useStyles = () => {
  const theme = useTheme();
  const styles = StyleSheet.create({
    container: {
      flex: 1,
      paddingBottom:5,
      justifyContent: 'center',
      alignItems: 'center',
      backgroundColor: theme.colors.accent,

    },
    Musiccontainer: {
      backgroundColor: theme.colors.accent,
    },


  });
  return styles;
}
export default useStyles;

