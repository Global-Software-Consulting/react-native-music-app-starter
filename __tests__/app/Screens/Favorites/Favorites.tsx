/**
 * @format
 */
import 'react-native';
import React from 'react';
import Favorites from '../../../../app/screens/Favorites';
// Note: test renderer must be required after react-native.
import { render } from '@testing-library/react-native';

jest.mock("react-native-responsive-screen", () => {
    return {
        heightPercentageToDP: jest.fn(),
        widthPercentageToDP: jest.fn(),

    }


});
jest.mock('@react-navigation/native', () => {
    return {
        useIsFocused: jest.fn(),
        useNavigation: jest.fn()
    }


});
jest.mock('react-i18next', () => {
    return {
        useTranslation: () => {
            return {
                t: jest.fn(),
            };
        },
    };
});

jest.mock('react-native-iphone-x-helper', () => ({
    getStatusBarHeight: jest.fn(),
    getBottomSpace: jest.fn(),
}));
jest.mock('react-redux', () => ({
    useDispatch: jest.fn(),
    useSelector: jest.fn(),
}));
describe("group tesing goes here", () => {
    test('renders correctly', () => {
        const tree = render(<Favorites />);
        expect(tree).toMatchSnapshot();
    });
})

