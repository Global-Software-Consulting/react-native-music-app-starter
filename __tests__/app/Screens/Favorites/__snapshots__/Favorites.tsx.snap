// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`group tesing goes here renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#03dac4",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#03dac4",
        "borderBottomColor": "lightgray",
        "elevation": 5,
        "flexDirection": "row",
        "height": undefined,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.25,
        "shadowRadius": 3.84,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
          "paddingHorizontal": 10,
        }
      }
    >
      <mockedDefaultExport
        name="keyboard-arrow-left"
        size={30}
        style={
          Object {
            "color": "#6200ee",
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "alignSelf": "center",
          "flex": 1,
          "justifyContent": "center",
          "marginRight": 25,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "center",
            "marginRight": 5,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              Object {
                "textAlign": "left",
              },
              Object {
                "fontSize": undefined,
                "fontWeight": "bold",
              },
            ]
          }
        />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#03dac4",
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#03dac4",
          "flexDirection": "row",
          "padding": 15,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              Object {
                "textAlign": "left",
              },
              Object {
                "fontSize": undefined,
                "fontWeight": "bold",
              },
            ]
          }
        />
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#03dac4",
          "flex": 1,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            Object {
              "textAlign": "left",
            },
            Object {
              "color": "darkgray",
              "fontSize": undefined,
              "justifyContent": "center",
              "textAlign": "center",
            },
          ]
        }
      />
    </View>
  </View>
</View>
`;
